components/engine_traits/src/sst.rs

```diff
@@ -59,6 +60,9 @@ where
     /// Set it to true, the builder builds a in-memory SST builder.
     fn set_in_memory(self, in_memory: bool) -> Self;
 
+    /// set other config specified by writer
+    fn set_conf_ext(self, conf: E::SstWriterConfExt) -> Self;
```

Comment:
How about just `set_compression`? So we can avoid adding a new associated type.

Reply:
Seems compression is a engine specified  config. I'm not sure if a method `set_compression` is meaningful for other engine?

Reply:
@brson  how do you think about this?

Reply:
I have reconsidered this, since it's a SstExt, so it should be always ok to set compression type for the sst file, so I will change this to `set_compression` and remove the `SstExt:: SstWriterConfExt `